File: 09/CS/CustomControls/09CustomControls/Client/UserCode/CustomButton912.cs
using System;
using System.Linq;
using System.IO;
using System.IO.IsolatedStorage;
using System.Collections.Generic;
using Microsoft.LightSwitch;
using Microsoft.LightSwitch.Framework.Client;
using Microsoft.LightSwitch.Presentation;
using Microsoft.LightSwitch.Presentation.Extensions;
namespace LightSwitchApplication
{
    public partial class CustomButton912
    {
        partial void SaveData_Execute()
        {
            // Write your code here.
            this.Save();
            this.ShowMessageBox("Saved");
        }


        partial void CustomButton912_InitializeDataWorkspace(global::System.Collections.Generic.List<global::Microsoft.LightSwitch.IDataService> saveChangesTo)
        {
            // Write your code here.
            this.CustomerProperty = new Customer();
        }

        partial void CustomButton912_Saved()
        {
            // Write your code here.
            //this.Close(false);
            Application.Current.ShowDefaultScreen(this.CustomerProperty);
        }

    }
}

File: 09/VB/09CustomControls/Client/UserCode/CustomButton912.vb

Namespace LightSwitchApplication

    Public Class CustomButton912

        Private Sub CustomButton912_InitializeDataWorkspace(ByVal saveChangesTo As Global.System.Collections.Generic.List(Of Global.Microsoft.LightSwitch.IDataService))
            ' Write your code here.
            Me.CustomerProperty = New Customer()
        End Sub

        Private Sub CustomButton912_Saved()
            ' Write your code here.
            'Me.Close(False)
            Application.Current.ShowDefaultScreen(Me.CustomerProperty)
        End Sub

        Private Sub SaveData_Execute()
            ' Write your code here.
            Me.Save()
            ' You can call some some other logic here...
            ShowMessageBox("Data Saved")

        End Sub
    End Class

End Namespace